### PR TITLE
Implode pipe for arrays

### PR DESCRIPTION
**Pull request by SBCDevelopers**
Does the same thing as implode in PHP and more because it can implode both simple array and array of objects by specifying the `delimiter` and `objectKey`  that you want to implode by.

### CODE REVIEW

File: src/pipes/array/implode [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    if(!Array.isArray(value)){
+      let _type = typeof value;
+      throw new Error(`Implode expect an array ${_type} given!`);
+    }
+    
+    if(delimiter == undefined)
+      throw new Error('You need to specify the delimiter!');
+    
+    console.log(value);
```
Comment (jeroenvheel) [inline src/pipes/array/implode]: console.log should be removed

File: src/pipes/array/implode [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export class ImplodePipe implements PipeTransform {
+
+  transform(value: any[], delimiter: any, objectKey?: string): string {
+
+    if(!Array.isArray(value)){
+      let _type = typeof value;
+      throw new Error(`Implode expect an array ${_type} given!`);
+    }
+    
+    if(delimiter == undefined)
```
Comment (danrevah) [inline src/pipes/array/implode]: use `isUndefined()` helper

File: src/pipes/array/implode (hunk 1 above)
Comment (danrevah) [reply src/pipes/array/implode]: yep

File: src/pipes/array/implode [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      throw new Error('You need to specify the delimiter!');
+    
+    console.log(value);
+    // if objectKey is not specified then it's a simple array
+    if(objectKey == undefined){
```
Comment (danrevah) [inline src/pipes/array/implode]: just 
```typescript
const arr = isUndefined(key) ? value : value.map(e => e[key]) 
return arr.join(delimiter);
```

File: src/pipes/array/implode [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+// does the same thing as implode in PHP
```
Comment (danrevah) [inline src/pipes/array/implode]: remove comments

File: src/pipes/array/implode [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// concat elements of array and return string
+@Pipe({
+  name: 'implode'
+})
+export class ImplodePipe implements PipeTransform {
+
+  transform(value: any[], delimiter: any, objectKey?: string): string {
+
+    if(!Array.isArray(value)){
+      let _type = typeof value;
```
Comment (danrevah) [inline src/pipes/array/implode]: remove

File: src/pipes/array/implode [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      throw new Error(`Implode expect an array ${_type} given!`);
```
Comment (danrevah) [inline src/pipes/array/implode]: change to `throw new Error(`Implode expected an array but ${typeof value} given.`)`

**Comment by danrevah** [comment]
FYI - if this stays without a response for 2 week's it will be closed.

**Pull request closed without merging**